client.go

```diff
@@ -211,22 +211,20 @@ func (c *client) listen() {
 		}
 		data = data[:n]
 
-		err = c.handlePacket(addr, data)
-		if err != nil {
-			utils.Errorf("error handling packet: %s", err.Error())
-			c.session.Close(err)
-			break
-		}
+		c.handlePacket(addr, data)
 	}
 }
 
-func (c *client) handlePacket(remoteAddr net.Addr, packet []byte) error {
+func (c *client) handlePacket(remoteAddr net.Addr, packet []byte) {
 	rcvTime := time.Now()
 
 	r := bytes.NewReader(packet)
 	hdr, err := ParsePublicHeader(r, protocol.PerspectiveServer)
 	if err != nil {
-		return qerr.Error(qerr.InvalidPacketHeader, err.Error())
+		utils.Errorf("error handling packet: %s", err.Error())
+		// drop this packet if we can't read parse the Public Header
+		// this packet could be malformed, or an injected packet with a spoofed remote address
+		return
```

Comment:
We can put it back on every code path that doesn't enter the session. Putting it back is never required though, just a (small) matter of performance.